crates/topos-tce-storage/src/validator/mod.rs

```diff
@@ -40,6 +50,59 @@ impl ValidatorStore {
     pub fn count_pending_certificates(&self) -> Result<usize, StorageError> {
         Ok(self.pending_tables.pending_pool.iter()?.count())
     }
+    pub fn get_pending_certificates(
+        &self,
+    ) -> Result<Vec<(PendingCertificateId, Certificate)>, StorageError> {
+        Ok(self.pending_tables.pending_pool.iter()?.collect())
+    }
+
+    pub fn multi_insert_pending_certificate(
+        &self,
+        certificates: &[Certificate],
+    ) -> Result<Vec<PendingCertificateId>, StorageError> {
+        let id = self
+            .pending_tables
+            .next_pending_id
+            .fetch_add(certificates.len() as u64, Ordering::Relaxed);
```

Comment:
I'm confused here: if `next_pending_id` is, say, 5 and I have a `certificates` with, say 3 certs in it, then won't this code put `8` in `id`? And below the `ids` will contain `8, 9 ,10`? Why have gaps in the ids? I would have thought we'd want `6, 7, 8` in the `ids` at the end of it all.

Reply:
`next_pending_id` represent the `id` of the next pending. In your example:

```
certificates = [A, B, C]
next_pending_id = 5
-> A -> pending_id = 5
-> A -> pending_id = 6
-> A -> pending_id = 7
next_pending_id becomes = 8
```

The `fetch_add` methods returns the old value then when reserving the 3 slots you get the starting point.